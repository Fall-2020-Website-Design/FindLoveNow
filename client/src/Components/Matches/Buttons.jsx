import React, { Component } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Image from 'react-bootstrap/Image';
import Button from 'react-bootstrap/Button';

import checkmark_image from '../../Images/checkmark.svg';
import x_image from '../../Images/x.png';
import previous_image from '../../Images/prev.svg';

export class Buttons extends Component {
    render() {
        return (
            <div>
                <Col className="d-inline-flex justify-content-center">
                    <Row className="d-inline-flex justify-content-center btn-row pt-4 pl-4">
                        <Col>
                            <Button variant="secondary" className="btn-circle-xl btn-xl ml-5 btn-check mr-2" onClick={this.props.accept} >
                                <Image alt="" src={checkmark_image} className=" align-top" fluid />
                            </Button>
                        </Col>

                        <Col className="pt-3 pl-5">
                            <Button variant="secondary" className="btn-circle-sm btn-sm" onClick={this.props.prev} >
                                <Image alt="" src={previous_image} className="align-top pt-1" fluid />
                            </Button>
                        </Col>

                        <Col>
                            <Button variant="secondary" className="btn-circle-xl btn-xl mr-5" onClick={this.props.reject} >
                                <Image alt="" src={x_image} className="align-top" fluid />
                            </Button>
                        </Col>
                    </Row>
                </Col>
            </div>
        )
    }
}

export default Buttons
