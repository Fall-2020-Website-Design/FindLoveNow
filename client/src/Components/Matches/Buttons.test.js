import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Buttons } from './Buttons';


Enzyme.configure({ adapter: new Adapter() });

describe('Buttons', () => {

	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Buttons  />);
    });
    
    test('renders', () => {
		const component = renderer.create(
			<MemoryRouter>
				<Buttons  />
			</MemoryRouter>
		).toJSON();
		
		expect(component).toMatchSnapshot();
    });
    

});