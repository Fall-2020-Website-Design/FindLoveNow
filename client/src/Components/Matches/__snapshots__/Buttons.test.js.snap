// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders 1`] = `
<div>
  <div
    className="d-inline-flex justify-content-center button-row col"
  >
    <div
      className="d-inline-flex justify-content-center btn-row pt-4 buttons-width row"
    >
      <div
        className="mr-5 col"
      >
        <button
          className="btn-circle-xl btn-xl btn-check btn btn-secondary"
          disabled={false}
          type="button"
        >
          <img
            alt=""
            className=" align-top img-fluid"
            src="checkmark.svg"
          />
        </button>
      </div>
      <div
        className="pt-3 ml-3 col"
      >
        <button
          className="btn-circle-sm btn-sm btn btn-secondary"
          disabled={false}
          type="button"
        >
          <img
            alt=""
            className="align-top pt-1 img-fluid"
            src="prev.svg"
          />
        </button>
      </div>
      <div
        className="ml-5 col"
      >
        <button
          className="btn-circle-xl btn-xl btn btn-secondary"
          disabled={false}
          type="button"
        >
          <img
            alt=""
            className="align-top img-fluid"
            src="x.png"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
